src/libstore/ssh.cc

```diff
@@ -111,8 +111,10 @@ std::unique_ptr<SSHMaster::Connection> SSHMaster::startCommand(const std::string
             reply = readLine(out.readSide.get());
         } catch (EndOfFile & e) { }
 
-        if (reply != "started")
```

Comment:
I see the code you updated both times was already duplicated

Can you deduplicate it with a lambda?

```c++
auto checkStarted = [&]() {
        std::string reply;
        try {
            reply = readLine(out.readSide.get());
        } catch (EndOfFile & e) { }

        if (reply != "started") {
            printTalkative("SSH output first line: %s", reply);
            throw Error("failed to start SSH connection to '%s'", host);
        }
};
```

Reply:
I think you missed that both spots are in different functions, so a lambda won't work.  
Also the second one copy has slightly different messages (includes "master") to help differentiation.